x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsState.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.dataframe;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Locale;
+
+public enum DataFrameAnalyticsState implements Writeable {
+
+    STARTED, REINDEXING, ANALYZING, STOPPING, STOPPED;
```

Comment:
Keeping the persistent task around with a `FAILED` status is really useful for working out which node to ask a customer to send logs from to investigate a failure (see also https://github.com/elastic/elasticsearch/issues/30006#issuecomment-448235425).  If we remove the persistent task for failures then we have no way of knowing which node the task was running on at the time it failed other than requesting logs for _all_ nodes and searching through them for the failure messages.  And getting logs from _all_ nodes may be extremely time-consuming for a customer, so they might not do it, leading to a poor support experience.

On the other hand, having a `FAILED` status means we need a way to eventually remove the failed persistent task.  For anomaly detectors we have force-close for this.  We would need an equivalent for analytics tasks.

Reply:
It might be 2 different things, `FAILED` could mean a problem in the C++ backend or a broken configuration and not a problem with the persistent task (framework).

Reply:
Yes, I don't think a `FAILED` status would ever mean a problem with the persistent task framework, as persistent tasks are really just documents in cluster state.  It's for the C++ problems that keeping the persistent task in existence with status `FAILED` is really useful, as the full set of log messages from the C++ will be in the log file of the node that ran the C++ process.  Similarly if the failure is with starting the C++ process it's invaluable to know which node that happened on.

Reply:
To make this more clear: I am _not_ questioning the usefulness of `FAILED` in persistent tasks.

I am raising the question whether the analytics state (and transforms as well) should have it's own `FAILED` state _in addition_.

We recently discussed making the creation of the persistent task more lazy, meaning not creating the task on PUT, but only after calling `_start`. That decouples it from persistent tasks and means an analysis/transform is not only a persistent task but more than that. 

Lets look at an example: let's say the source data frame gets deleted _after_ the analysis has been created but before you call `_start`. The persistent task does not exist at this point in time. It makes sense to be able to put the transform/analysis into a `FAILED` state.

Reply:
This set of states is not complete and it will get to its final shape as we address operational sides. There's nothing setting the state to `FAILED` yet (though the same could be said for the `STOPPING` state). I think it'd be fine to add it when we implement something that needs the state.

Reply:
👍 for setting a failed in the future.